File: Core.DomainModel/Shared/YesNoIrrelevantOption.cs
using System;

namespace Core.DomainModel.Shared
{
    public enum YesNoIrrelevantOption
    {
        NO = 0,
        YES = 1,
        IRRELEVANT = 2,
        UNDECIDED = 3
    }

    public static class YesNoIrrelevantOptionExtensions
    {
        public static string GetReadableName(this YesNoIrrelevantOption yesNoIrrelevantOption)
        {
            return yesNoIrrelevantOption switch
            {
                YesNoIrrelevantOption.NO => "Nej",
                YesNoIrrelevantOption.YES => "Ja",
                YesNoIrrelevantOption.IRRELEVANT => "Ikke relevant",
                YesNoIrrelevantOption.UNDECIDED => "",
                _ => throw new InvalidOperationException($"Invalid yesNoIrrelevantOption value: {yesNoIrrelevantOption}"),
            };
        }
    }
}

File: Presentation.Web/app/models/api/Shared/YesNoIrrelevantOption.ts
module Kitos.Models.Api.Shared {
    export enum YesNoIrrelevantOption {
        NO = 0,
        YES = 1,
        IRRELEVANT = 2,
        UNDECIDED = 3
    }

    export class YesNoIrrelevantOptionMapper {
        static getValueToTextMap() {
            return Object
                .keys(YesNoIrrelevantOption)
                .filter(k=>isNaN(parseInt(k)) === false)
                .reduce((acc, next, _) => {
                    const text = Models.ViewModel.Shared.YesNoIrrelevantOptions.getText(parseInt(next));

                    //Set by numeric and text value
                    acc[next] = text;
                    acc[YesNoIrrelevantOption[next]] = text;
                    return acc;
                },
                    {}
                );
        }
    }
}